### PR TITLE
bug fix for cmul

### PR DESCRIPTION
**Pull request by zhangxiaoli73**
## What changes were proposed in this pull request?

fix cmul error when batchsize equals one of element number. (e.g. input size (64, 64, 112, 112))

## How was this patch tested?

unit tests and example tests

## Related links or issues (optional)
fixed https://github.com/intel-analytics/BigDL/issues/2812

### CODE REVIEW

File: spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/CMul.scala [hunk 1]
```diff
@@ -82,13 +82,19 @@ class CMul[T: ClassTag](
...
 
   private def mulOneDimWeight(dim: Int, expand: Tensor[T], output: Tensor[T]): Unit = {
-    val multiplyDimSize = expand.size(dim)
-    var dimOfOutput : Int = 1
-    while(output.size(dimOfOutput) != multiplyDimSize) {
-
-      dimOfOutput += 1
-      require(dimOfOutput <= output.dim(), s"OutOfBound : " +
-        s"Output does not have a dimension of $multiplyDimSize elements")
+    var dimOfOutput : Int = dim
```
Comment (wzhongyuan) [inline spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/CMul.scala]: `outputDim` ?

File: spark/dl/src/test/scala/com/intel/analytics/bigdl/nn/CMulSpec.scala [hunk 2]
```diff
@@ -65,6 +66,22 @@ class CMulSpec extends FlatSpec with Matchers {
     gradInput3 should be (gradInput4)
     layer2.gradWeight should be (layer1.gradWeight.mul(0.5))
   }
+
+  "CMUl" should "work good on batch input" in {
```
Comment (wzhongyuan) [inline spark/dl/src/test/scala/com/intel/analytics/bigdl/nn/CMulSpec.scala]: works or works well

Comment (wzhongyuan) [inline spark/dl/src/test/scala/com/intel/analytics/bigdl/nn/CMulSpec.scala]: it's for special batch size right ?

File: spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/CMul.scala (hunk 1 above)
Comment (zhangxiaoli73) [reply spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/CMul.scala]: ok, changed

File: spark/dl/src/test/scala/com/intel/analytics/bigdl/nn/CMulSpec.scala (hunk 2 above)
Comment (zhangxiaoli73) [reply spark/dl/src/test/scala/com/intel/analytics/bigdl/nn/CMulSpec.scala]: ok, changed

Comment (zhangxiaoli73) [reply spark/dl/src/test/scala/com/intel/analytics/bigdl/nn/CMulSpec.scala]: yes, e.g.  batchsize equals channel number, then there will be wrong result.

**Comment by zhangxiaoli73** [comment]
jenkins pass： http://mlt-bddl01.sh.intel.com:8080/job/BigDL-PRVN-lite/637/

**Pull request merged**